File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/model/items/ItemSignal.java
package de.limbusdev.guardianmonsters.model.items;

import com.badlogic.ashley.signals.Signal;

/**
 * ItemSignal
 *
 * @author Georg Eckert 2017
 */

public class ItemSignal extends Signal<Item> {

    public final Item item;
    public final Message message;

    public enum Message {
        DELETED, ADDED,
    }

    public ItemSignal(Item item, Message message) {
        this.item = item;
        this.message = message;
    }
}

File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/items/ItemSignal.kt
package de.limbusdev.guardianmonsters.guardians.items

import com.badlogic.ashley.signals.Signal

/**
 * ItemSignal
 *
 * @author Georg Eckert 2019
 */

class ItemSignal
(
        val item: Item,
        val message: Message
)
    : Signal<Item>()
{
    enum class Message { DELETED, ADDED }
}
